components/style/values/generics/grid.rs

```diff
@@ -505,6 +487,46 @@ impl<L: ToComputedValue> ToComputedValue for TrackRepeat<L> {
     }
 }
 
+/// Track list values. Can be <track-size> or <track-repeat>
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum TrackListValue<T> {
+    /// A <track-size> value.
+    TrackSize(TrackSize<T>),
+    /// A <track-repeat> value.
+    TrackRepeat(TrackRepeat<T>),
+}
+
+impl<L: ToCss> ToCss for TrackListValue<L> {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        match *self {
+            TrackListValue::TrackSize(ref size) => size.to_css(dest),
+            TrackListValue::TrackRepeat(ref repeat) => repeat.to_css(dest),
+        }
+    }
+}
+
+impl<L: ToComputedValue> ToComputedValue for TrackListValue<L> {
+    type ComputedValue = TrackListValue<L::ComputedValue>;
+
+    #[inline]
+    fn to_computed_value(&self, context: &Context) -> Self::ComputedValue {
+        match *self {
+            TrackListValue::TrackSize(ref size) => TrackListValue::TrackSize(size.to_computed_value(context)),
+            _ => unreachable!("Should only compute track-size values"),
```

Comment:
hmm... this looks somewhat suspicious... Maybe something should be refactored here? Maybe it's worth to just make it idempotent (in which case you can use `derive(ToComputedValue)`.